src/hooks/__tests__/useApi.test.ts

```diff
@@ -0,0 +1,44 @@
+import { renderHook } from '@testing-library/react-hooks';
+
+import useApi from '../useApi';
+
+describe('useApi', () => {
+  it('returns empty error and result while fetching', () => {
+    // we use a promise here that never returns anything, so the hook never returns a result
+    jest.spyOn(window, 'fetch').mockImplementationOnce(() => new Promise<Response>(() => { }));
+    const { result } = renderHook(() => useApi('/foo/loading'));
+    const [error, apiResult] = result.current;
+
+    expect(error).toBeNull();
+    expect(apiResult).toBeNull();
+  });
+
+  it('returns response JSON if request succeeds', async () => {
+    const expectedResult = { test: 42 };
+    const response = new Response(JSON.stringify(expectedResult));
+    jest.spyOn(window, 'fetch').mockResolvedValue(response);
+
+    const { result, waitForNextUpdate } = renderHook(() => useApi('/foo/success'));
+
+    await waitForNextUpdate();
+
+    const [error, apiResult] = result.current;
+
+    expect(error).toBeNull();
+    expect(apiResult).toStrictEqual(expectedResult);
+  });
+
+  it('returns an error if request fails', async () => {
+    const expectedError = new Error('Expected error');
+    jest.spyOn(window, 'fetch').mockRejectedValue(expectedError);
```

Comment:
Similarly to the `mockResolvedValue()` above, we use `mockRejectedValue()` here to throw an error in the `try-catch` block of the `useApi` hook. This way we can test that errors are returned as expected.